ipapython/certdb.py

```diff
@@ -791,7 +791,7 @@ def add_cert(self, cert, nick, flags):
         self.run_certutil(args, stdin=cert.public_bytes(x509.Encoding.PEM))
 
     def delete_cert(self, nick):
-        self.run_certutil(["-D", "-n", nick])
+        self.run_certutil(["-F", "-n", nick])
```

Comment:
``-F`` only deletes the private key. We want to delete both.

from man certutil
> When you delete keys, be sure to also remove any certificates associated with those keys from the certificate database, by using -D. Some smart cards do not let you remove a public key you have generated. In such a case, only the private key is deleted from the key pair. You can display the public key with the command certutil -K -h tokenname.

Reply:
@tiran Very Thanks for Review.

But I believe '-F' option deletes both keys & certificates. 

This is the test I did on local setup.
//Generate CA Key
    `# openssl genrsa -aes256 -out private/rootCA.key 4096;    chmod 400 private/rootCA.key`     

//Generate CA CERT
    `# openssl req -config openssl.cnf -key private/rootCA.key  -new -x509 -days 7300 -sha256 -extensions v3_ca -out certs/rootCA.crt`    

//Looking inside Certificate DB
`# mkdir nssdb-test;    cd nssdb-test;    certutil -N -d .`
`# ls -ltr`
total 60
-rw-------. 1 root root 16384 Jan  8 23:30 secmod.db
-rw-------. 1 root root 16384 Jan  8 23:30 key3.db
-rw-------. 1 root root 65536 Jan  8 23:30 cert8.db

//Create 1 File 'server.pfx' containing both Key and certificate
`# openssl pkcs12 -export     -out server.pfx     -inkey ../private/rootCA.key     -in ../certs/rootCA.crt`       
Enter pass phrase for ../private/rootCA.key:
Enter Export Password:
Verifying - Enter Export Password:
`#`

//Add pkcs12/*.pfx file to certificate DB
`# pk12util -i server.pfx     -d .`            
Enter password for PKCS12 file: 
pk12util: no nickname for cert in PKCS12 file.
pk12util: using nickname: Default Company Ltd
pk12util: PKCS12 IMPORT SUCCESSFUL

//Verifying Cert, Key is added to DB
`# certutil -L -d .`
Certificate Nickname        Trust Attributes
                                         SSL,S/MIME,JAR/XPI
Default Company Ltd           u,u,u                <<<<<<<<<Cert Added to DB
`# certutil  -K  -d.`          <<<<<<Key added to DB
certutil: Checking token "NSS Certificate DB" in slot "NSS User Private Key and Certificate Services"
< 0> rsa      9c962813d1b7619f1376e2fa5ea14203bf6fb4ca   Default Company Ltd

//Trying to use -D -F
`# certutil -F -D -n 'Default Company Ltd'     -d .`      
certutil: only one command at a time!              <<<<<<<<<<<<<<<<
You entered:  -D -F 
`#`

//Deleting Key, cert with -F
`# certutil -F -n 'Default Company Ltd'     -d .`

//Verifying Cert, Key are deleted from Certificate DB
`# certutil -L -d .`
Certificate Nickname                                         Trust Attributes
                                                             SSL,S/MIME,JAR/XPI
`# certutil -K -d .`
certutil: Checking token "NSS Certificate DB" in slot "NSS User Private Key and Certificate Services"
certutil: no keys found
`#`

As we can see -F option deletes key+associated certificate from Certificate DB.

Kindly do correct me.. If my interpretation is wrong...